src/azure-cli/azure/cli/command_modules/role/_help.py

```diff
@@ -115,6 +115,11 @@
     The output includes credentials that you must protect. Be sure that you do not include these credentials
     in your code or check the credentials into your source control. As an alternative, consider using
     [managed identities](https://aka.ms/azadsp-managed-identities) if available to avoid the need to use credentials.
+    
+    **Note:** This command resets the credentials of the Service Principal, not the App Registration.
+    As such, the credential shown in the output will be different from the credential shown in the portal under the App Registration pane.
+    Credentials of a Service Principal are not visable in the portal, only via the az cli or Microsoft Graph API.
```

Comment:
We don't want to describe the behavior of Azure Portal since Azure Portal are subject to change. It's hard to maintain. Once Azure Portal changes its behavior, CLI's document will be outdated.